arduino/splitflap/esp32/splitflap_task.cpp

```diff
@@ -265,6 +276,15 @@ SplitflapState SplitflapTask::getState() {
     return state_cache_;
 }
 
+void SplitflapTask::writeBuffer(const char *word) {
+    SemaphoreGuard lock(semaphore_);
+    recv_count = NUM_MODULES;
+    for (int i = 0; i < NUM_MODULES; i++) {
+        recv_buffer[i] = word[i];
+    }
```

Comment:
Can you pass the length as an argument and use that to set `recv_count` and the loop to avoid accidentally reading invalid data if an MQTT message doesn't include enough characters?

Reply:
That's definitely better. Good catch

Reply:
Made this change in the latest commit.